sympy/functions/elementary/hyperbolic.py

```diff
@@ -1233,3 +1233,110 @@ def inverse(self, argindex=1):
 
     def _eval_rewrite_as_log(self, arg):
         return log(1/arg + sqrt(1/arg**2 - 1))
+
+
+class acsch(Function):
+    """
+    The inverse hyperbolic cosecant function.
+
+    * acsch(x) -> Returns the inverse hyperbolic cosecant of x
+
+    Examples
+    ========
+
+    >>> from sympy import acsch, sqrt, S
+    >>> from sympy.abc import x
+    >>> acsch(x).diff(x)
+    -1/(x**2*sqrt(1 + x**(-2)))
+    >>> acsch(1).diff(x)
+    0
+    >>> acsch(1)
+    log(1 + sqrt(2))
+    >>> acsch(S.ImaginaryUnit)
+    -I*pi/2
+    >>> acsch(-2*S.ImaginaryUnit)
+    I*pi/6
+    >>> acsch(S.ImaginaryUnit*(sqrt(6) - sqrt(2)))
+    -5*I*pi/12
+
+    References
+    ==========
+
+    .. [1] http://en.wikipedia.org/wiki/Hyperbolic_function
+    .. [2] http://dlmf.nist.gov/4.37
+    .. [3] http://functions.wolfram.com/ElementaryFunctions/ArcCsch/
+
+    """
+
+    def fdiff(self, argindex=1):
+        if argindex == 1:
+            z = self.args[0]
+            return -1/(z**2*sqrt(1 + 1/z**2))
+        else:
+            raise ArgumentIndexError(self, argindex)
+
+    @classmethod
+    def eval(cls, arg):
+        arg = sympify(arg)
+
+        if arg.is_Number:
+            if arg is S.NaN:
+                return S.NaN
+            elif arg is S.Infinity:
+                return S.Zero
+            elif arg is S.NegativeInfinity:
+                return S.Zero
+            elif arg is S.Zero:
+                return S.Infinity
+            elif arg is S.One:
+                return log(1 + sqrt(2))
+            elif arg is S.NegativeOne:
+                return - log(1 + sqrt(2))
+
+        if arg.is_number:
+            cst_table = {
```

Comment:
> I agree, but most of these are imaginary? Am I misunderstanding something?

I'm sorry I did this PR a long time ago. You are correct. This does seem to focus on Imaginary values. **I can change this if necessary?** Also since `acsch(I*x) = -I*acsc(x)`, that can be directly used. But `csch` of real angles are not very standard, so it would not be easy to implement. For e.g. `csch(pi/12) = -2*sinh(pi/12)/(1-cosh(pi/6))`